example/count.cpp

```diff
@@ -0,0 +1,38 @@
+
+// Range v3 library
+//
+//  Copyright Jeff Garland 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+// 
+
+
+///[count]
+// This example demonstrates counting the number of 
+// elements that match a given value. 
+// output...
+// vector:   2
+// array:    2
+
+#include <range/v3/algorithm/count.hpp>   //specific includes
+#include <vector>
+#include <iostream>
+using std::cout;
+
+int main() {
+
+  std::vector<int> v { 6, 2, 3, 4, 5, 6 };
+  int c = ranges::count( v, 6 );
```

Comment:
`ranges::count` returns `difference_type_t`, so `long` may be more appropriate in here instead of `int` to avoid conversion warnings

Reply:
`auto` would be a superior choice: the difference type of `vector` iterators is implementation-defined. It's going to be `ptrdiff_t` on any sane implementation, which is equivalent to `int` or `long` or `long long` on most implementations. Choosing too small a type results in a warning here, choosing too large a type results in conversion warnings elsewhere when `c` is converted to a shorter type.

That said, `int` is dead simple here. If we want to turn off conversion warnings for examples I would not object.

Reply:
So let me say first that gcc-head gave me no warnings with the cmake.

That said I was also unaware of what it might be here.  So auto works fine for me, but I'm pondering adding a comment that explains the return type in the example.